app/controllers/order_items_controller.rb

```diff
@@ -4,11 +4,11 @@ class OrderItemsController < ApplicationController
   # more increases the quantity of an item in the cart
   def more # OrderItem.more <-- gimme more of this OrderItem
     if @item.product.has_available_stock?
-      @item.increment!(:quantity_ordered, 1)
+      @item.more!
     else
       # !W we should talk about whether this error message should be shown
       # and what it should truly say in production
-      flash[:error] = "You cannot increase the quantity of #{ @item.product.name } any further, because there are only #{ @item.quantity_ordered } in stock."
+      flash[:error] = @item.errors[:quantity_ordered]
```

Comment:
This would make more sense to me if it were just `@item.errors` in order to accommodate other errors that may happen.